### PR TITLE
Added archive db connection check with meaningful status instead of silent exit

### PR DESCRIPTION
**Pull request by dkijania**
Explain your changes:

Added simple routine to be more verbose regarding connection to archive database when launching local-network script.
Before it just silently exits and i had to jump to sources in order to learn why script exited. Now, script prints out information about connection attempt.

Explain how you tested your changes:

I did a manual tests with various cases:
- connection is correct
- database is down or bootstrapping
- parameters are invalid

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

File: scripts/mina-local-network/mina-local-network.sh [hunk 1]
```diff
@@ -181,6 +181,37 @@ spawn-archive-node() {
...
+check-db-connection() {
+  echo "Checking connection to archive database..."
+  printf "\n"
+
+  case `pg_isready -U ${PG_USER} -h ${PG_HOST} -p ${PG_PORT} -d ${PG_DB} &>/dev/null; echo $?` in
+    0)
+      echo "Connection to database successful!"
+      ;;
+    1)
+      echo -n "The server is rejecting connections (for example during startup)"
```
Comment (psteckler) [inline scripts/mina-local-network/mina-local-network.sh:195]: Why `-n`? The error message will be followed by the shell prompt.

File: scripts/mina-local-network/mina-local-network.sh [hunk 2]
```diff
@@ -181,6 +181,37 @@ spawn-archive-node() {
...
+      exit 1
+      ;;
+    2)
+      echo -n "There was no response from server to the connection attempt"
```
Comment (psteckler) [inline scripts/mina-local-network/mina-local-network.sh:199]: Should this message mention the database? Otherwise, hard to know which server is involved.

**Review by psteckler — CHANGES_REQUESTED** [review]
See comments.

File: scripts/mina-local-network/mina-local-network.sh (hunk 1 above)
Comment (dkijania) [reply scripts/mina-local-network/mina-local-network.sh]: removed

File: scripts/mina-local-network/mina-local-network.sh (hunk 2 above)
Comment (dkijania) [reply scripts/mina-local-network/mina-local-network.sh]: right!

**Review by psteckler — APPROVED** [review]

**Review by stevenplatt — APPROVED** [review]

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Pull request merged**